src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -37,7 +39,7 @@ public class WebJobsScriptHostService : IHostedService, IScriptHostManager, IDis
 
         public WebJobsScriptHostService(IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IScriptHostBuilder scriptHostBuilder, ILoggerFactory loggerFactory, IServiceProvider rootServiceProvider,
             IServiceScopeFactory rootScopeFactory, IScriptWebHostEnvironment scriptWebHostEnvironment, IEnvironment environment,
-            HostPerformanceManager hostPerformanceManager, IOptions<HostHealthMonitorOptions> healthMonitorOptions)
+            HostPerformanceManager hostPerformanceManager, IOptions<HostHealthMonitorOptions> healthMonitorOptions, IFunctionsSyncManager functionsSyncManager)
```

Comment:
Is there a good reason to add this logic to the existing service as opposed to introducing a new `IHostedService`? Feels like that would give us a cleaner separation.

Reply:
My only last question after the last round of changes is about this comment. Any downside on having this isolated in a different hosted service?